### PR TITLE
[MINOR][DOCS] ForeachBatch java example fix

### PR DESCRIPTION
**Pull request by roland1982**
### What changes were proposed in this pull request?
ForEachBatch Java example was incorrect

### Why are the changes needed?
Example did not compile

### Does this PR introduce any user-facing change?
Yes, to docs.

### How was this patch tested?
In IDE.

### CODE REVIEW

File: docs/structured-streaming-programming-guide.md [hunk 1]
```diff
@@ -2202,7 +2202,7 @@ streamingDatasetOfString.writeStream.foreach(
 In Java, you have to extend the class `ForeachWriter` ([docs](api/java/org/apache/spark/sql/ForeachWriter.html)).
 {% highlight java %}
 streamingDatasetOfString.writeStream().foreach(
-  new ForeachWriter[String] {
+  new ForeachWriter<String>() {
```
Comment (HyukjinKwon) [inline docs/structured-streaming-programming-guide.md]: This seems already fixed at https://github.com/apache/spark/commit/9a2cec9b1ee3ba095461d9c07c9932107228d42e . Can you remove this in your PR?

**Review by HyukjinKwon — COMMENTED** [review]
I see this is a second PR from you https://github.com/apache/spark/pull/27671 where it fixes similar typos. Can you double check the examples and fix in batch?

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by HyukjinKwon** [comment]
Please read https://spark.apache.org/contributing.html and fix the PR title, and others accordingly.

**Comment by srowen** [comment]
@roland1982 is this the last one you have?

**Comment by roland1982** [comment]
As far as I can see, yes.

**Comment by srowen** [comment]
Merged to master/3.0/2.4

**Pull request closed without merging**